### DISCUSSION TITLE
Collabora and Talk not working after installation

### DISCUSSION THREAD

**Discussion by apobrt** [nextcloud/all-in-one#1196] [Questions] [answered]
**Steps to reproduce**

1.  Install Nextcloud AIO following the steps in `readme.md`, skipping domain validation with `-e SKIP_DOMAIN_VALIDATION=true`
2. Enable the Collabora and Talk containers from the AIO Interface.
3. Open any .docx or .odt file, and
4. Create a chat in Talk

**Expected behavior**

- Collabora/Nextcloud Office loads the file in the browser
- Be able to chat or make a video call with anyone

**Actual behavior**

- Nextcloud is extremely slow
- When opening documents, this error appears: "Document Loading failed. Failed to load Nextcloud Office - please try again later"
- When creating a chat, this error appears: "Failed to establish signaling connection. Retrying …"

**Host OS**

CentOS 8.6

**Nextcloud AIO version**

Nextcloud AIO v2.0.3

**Other valuable info**

- strictly followed the installation guide in `readme.md`, but the only way to make it work was to skip domain validation
- deleting the "High-performance backend" from Settings > Talk, the chats and video calls work. This made it possible to use the chats
- disabling the "Nextcloud Office" from the app manager, the performance improves by a lot
Obviously, this way, the document's management still not works

**Comment by szaimen** [comment]
Hi, are you running AIO behind a reverse proxy?

**Comment by szaimen** [comment]
Also, did you open all required ports?

See https://github.com/nextcloud/all-in-one#which-ports-are-mandatory-to-be-open-in-your-firewallrouter

**Comment by apobrt** [comment]
Hi, ✋ 

> Hi, are you running AIO behind a reverse proxy?

No. The server machine is freshly installed and there are no other applications running: only Docker with Nextcloud AIO



> Also, did you open all required ports?
> 
> See https://github.com/nextcloud/all-in-one#which-ports-are-mandatory-to-be-open-in-your-firewallrouter

Yes. We followed the documentation and opened all the required ones.
We checked also with portckecker.co: 
![2022-09-26_14-32](https://user-images.githubusercontent.com/45424365/192276952-1b7fc334-ad5f-44c9-ad2f-fdfdee4a2301.png)
![20220926_142752](https://user-images.githubusercontent.com/45424365/192276993-6b4aa5c6-cb48-4c7c-beb7-2b99842b074a.png)

**Comment by szaimen** [comment]
Sorry but I cannot reproduce this. Works over here. My guess would be that you need to disalbe SELinux or configure it correctly to make it work...

**Comment by apobrt** [comment]
Nope, we've just tried, but it didn't work.

Just to understand: what should be the correct configuration if everything was working?
These are the screenshots we see of the Collabora server and the Talk HPB 
![2022-09-26 17_12_47-Impostazioni](https://user-images.githubusercontent.com/45424365/192323258-59a7b698-41cc-4a89-a098-6b65029c2f0c.png)

![2022-09-26 17_14_38-Impostazioni](https://user-images.githubusercontent.com/45424365/192323329-83df2390-0ed1-4fd3-be68-8ec8f15bbc81.png)

**Reply by szaimen** [reply]
The values ahould be set automatically correctly when you restart the containers

**Reply by szaimen** [reply]
I guess I will popbably need the talk and collabora container logs but don't have high hopes that we will find a solution since I cannot reproduce this...

**Reply by apobrt** [reply]
**Talk logs**
https://bin.disroot.org/?d4865a8bab71cc8c#8Pna5KeTt7wNcVKCtRnyAsFt1sAo5s8jv16jxp94YTSq

**Collabora logs**
https://bin.disroot.org/?20cd54f9541f534e#BjRjo2aMzSxraMHM5oC3cngK4s8vtHun2WQRo4e1kGEP

Note: these logs are pasted after disabling the Collabora app and removing the Talk HPB

**Reply by szaimen** [reply]
As for the talk logs, they will be improved with https://github.com/nextcloud/all-in-one/pull/1203 in the next update. The collabora logs look good. So yeah, as supposed do the logs not really help finding a solution to this...

**Comment by szaimen** [comment]
Do you maybe have an unusual Domain infrastructure? E.g. using cloudflare or a kind of proxy in any way?

**Reply by apobrt** [reply]
Mmmm we're investigating, we'll give more info asap.

In the meantime, I'll add another question, to better understand (and make people understand): following a tutorial to install Nextcloud AIO, during server configuration, the hosts file was being modified.
Specifically, the rule
`127.0.1.1 <subdomain>.<domain>.it alias`
Does this have any impact? We have tried both with and without, and nothing seems to change

**Reply by szaimen** [reply]
> Does this have any impact? We have tried both with and without, and nothing seems to change

yeah, I don't think it makes any difference

**Reply by apobrt** [reply]
So, this should be our infrastructure:

- we have a server and a domain on OVH
- under our domain, we created an A record for the subdomain

![2022-09-27_14-37](https://user-images.githubusercontent.com/45424365/192528251-4a82ce40-e5e1-4038-bfe0-c5686b198474.png)

- there is a firewall on that server, where we open and redirects the traffic from the FQDN to the server on the ports specified at https://github.com/nextcloud/all-in-one#explanation-of-used-ports

![2022-09-27_14-31](https://user-images.githubusercontent.com/45424365/192527309-0e02308e-b4ca-4bc6-8e41-631cbf410932.png)

- we do not use Cloudflare 
- we do not have any proxy

**Reply by apobrt** [reply]
All the firewall rules have the Intrusion Detection and Prevention System (IPS) enabled (the screenshot is slightly old, now also the port 8443 has it): it could be a problem?

**Reply by szaimen** [reply]
yes, that could be a problem...

**Reply by apobrt** [reply]
It doesn't seem to work: we have disabled for all the rules, but noting has changed. 
I followed the steps at https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance, but after entering the AIO interface, I restore the installation using the backup : it could be possible that restoring the installation from the backup also restores the network configurations?
I'm now a little discouraged, we've tried almost anything:

- The skip domain flag (necessary to go further)
- The tables used by firewalld (https://github.com/nextcloud/all-in-one#how-to-resolve-firewall-problems-with-fedora-linux-rhel-os-centos-suse-linux-and-others)
- The Seccomp feature (but it's enabled in our kernel)
- Disable SELinux
- The firewall rules configuration

It's possible that is something related to the OS? Are there anyone that succeeded using CentOS?
Thanks for your time

**Reply by szaimen** [reply]
> It's possible that is something related to the OS? Are there anyone that succeeded using CentOS?
> Thanks for your time

Good questions. I must admin I have not tried it on CentOS yet. One additional idea would how you installed docker..  Apart from that I am out of ideas, I am afraid!

**Reply by apobrt** [reply]
We couldn't install Docker via the convenience script, because it was failing the process. On one forum (I can't find anymore the link), someone recommended modifying the shell script by replacing in the `get_distribution` method "$ID" with "$ID_LIKE", but even that didn't work.
We then followed the official documentation: https://docs.docker.com/engine/install/centos/#install-using-the-repository

At this point, we just have to try changing the OS: do you recommend any in particular? The online tutorials mostly deal with Ubuntu, so we were thinking of using that one

**Reply by apobrt** [reply]
> I restore the installation using the backup : it could be possible that restoring the installation from the backup also restores the network configurations?

If we redo the complete installation from scratch, without going through the backup, would it change anything?

**Reply by szaimen** [reply]
Ubuntu is what we are using on our test instance so it is indeed well tested and should work.

**Reply by apobrt** [reply]
Hi.
We've tried deleting all and installing Ubuntu 20.04, but nothing changed: we had to do the skip domain validation, and Collabora and Talk are still not working.
We have some question, to understand better :

1. it is possible that these issues are related to the skip-domain validation?
2. we have not mentioned before, but under Settings > Overview we see this warning: 

> This server has no working internet connection: Multiple endpoints could not be reached. This means that some of the features like mounting external storage, notifications about updates or installation of third-party apps will not work. Accessing files remotely and sending of notification emails might not work, either. Establish a connection from this server to the internet to enjoy all features.

Could be related?

3. the ports listed here https://github.com/nextcloud/all-in-one#which-ports-are-mandatory-to-be-open-in-your-firewallrouter are **the only ones** that should be opened? Our firewall filters all the others

Thanks!

**Reply by szaimen** [reply]
Hm... I have the gut feeling that still something with your firewall configuration does not work as expected if it doesnt even work om ubuntu for you.

**Reply by szaimen** [reply]
Does your firewall allow websocket connections?

**Reply by apobrt** [reply]
> Does your firewall allow websocket connections?

We checked, but it doesn't appear that our firewall has an option to allow or disable websocket connections: isn't it enough that the HTTP and HTTP ports are open (which we did)? How can we check that they are allowed?
Also: websocket connections between servers/containers happen within the same server, so why would they go through the firewall?

**Reply by szaimen** [reply]
All the parts that are not working for you need websocket connections to the outside working. So there seem to be a pattern indeed. You can e.g. look at the browser logs when opening a document in Nextcloud Office.

**Reply by apobrt** [reply]
We don't find a specific rule in our firewall to allow websocket connection, but we're investigating more.
In the meantime, here are the console output:

- when we try to open the default .docx document (it's not very helpful…)
![2022-10-07 14_14_01-Impostazioni](https://user-images.githubusercontent.com/45424365/194552766-ec00c49f-e0ee-4b7d-a428-783d78722a9e.png)

- when we try to set up the Nextcloud/Collabora server
![2022-10-07 14_20_38-Settings - Nextcloud – Mozilla Firefox](https://user-images.githubusercontent.com/45424365/194552956-ef2d18d2-5e93-42c7-924b-8e932ed74aae.png)

**Reply by apobrt** [reply]
We've changed some rules on our firewall, and now we have a new error, 502 bad gateway:
![NME4KV3Xl00xPhRL](https://user-images.githubusercontent.com/45424365/194884866-eaa93e17-b4ad-4ad2-87ff-d0c7cccd7db3.png)

**Comment by apobrt** [comment]
>     1. it is possible that these issues are related to the skip-domain validation?

Surfing online forums, we see others which have to skip the domain validation (eg: https://help.nextcloud.com/t/nextcloud-aio-install-fails-to-find-open-port-443-open/142867/12) and then cannot use Collabora and Talk like us: could this be a symptom of some "incorrect" configuration? What does the domain check verify? I have a feeling that if we could validate the domain, then everything would work

**Reply by szaimen** [reply]
No, validating the domain wouldn't have made it work. It simply check if some basic things are set up correctly but we do not check websocket connections.

**Reply by apobrt** [reply]
Oh, I see. Also, you already answered here, sorry: https://help.nextcloud.com/t/nextcloud-aio-setup-problem/141805/30

**Reply by apobrt** [reply]
We found this comment (https://help.nextcloud.com/t/nextcloud-aio-setup-problem/141805/48) which suggest considering to NAT the requests with the public address: so this means that every communication between Nextcloud and Collabora (and vice versa) pass through the public address? Even if they are on the same subnet (and, with AIO, on the same server)?

**Comment by szaimen** [comment]
Honestly it still looks like something is preventing websockets from working but otherwise no clue, sorry!

Everything works fine over here.

**Comment by szaimen** [comment, marked as answer]
Hey, we just had another idea to debug the problem. See https://github.com/nextcloud/all-in-one/discussions/1312#discussioncomment-3972346

**Comment by apobrt** [comment]
Hey,
some news on this topic:
TL;DR;
We couldn't manage to make it work; we changed the server, and opted for a simpler network configuration (without the external Endian firewall)
Now we have a Virtual Private Server hosted on OVH, no external firewall (only UFW on Ubuntu).

Thanks for all your support !!

**Comment by WaylonR** [comment]
Both these problems still exist.
Using this stack, behind a dockered nginx reverse proxy. Talk and collabora unusable.

**Reply by szaimen** [reply]
Hi, can you please open a new discussion for this? Thanks!

**Comment by wildernessfamily** [comment]
I had this exact same problem and was able to solve the problem. It may not be the same for all of you, but I thought I post how I solved the problem on my side. I use CloudFlare for my DNS. I'm not using Algo Tunnel for AIO. However, I do use CloudFlare Firewall including CloudFlare Bot Fight Mode feature. Reviewing CloudFlare WAF logs, I found that it was blocking the NextCloud Server Crawler. I created a rule to allow the AIO server IP. Once I whitelisted the AIO server IP all was good. I was able to connect Collabora Server and NC Talk.

**Comment by Lnx4F** [comment]
I'm having the same issues reported here regarding Talk only!

I have the same exact install process as @apobrt for step 1 (skipping domain validation). I have AIO Latest running on Ubuntu 22.04 Server on ESXI 7.0.

Talk will not work with Voice or Video, but text chat is fine. Starting a call from PC will ring. Answering call on other end will just sit there and nothing happens.

(I do not have Cloudflare or any special reverse-proxy setups in my home. Just a basic PFsense firewall with the required ports opened according to the all-in-one guide, and my own domain. My ISP is a dynamic IP, and I already have tools which auto-update my DNS)

Curiously, I followed the same troubleshooting step as @apobrt ("deleting the "High-performance backend" from Settings > Talk")

This made Talk Voice and Video calls work! But only locally. Calls over internet still fail. :(
(Port 3478 is open to the internet from my firewall)
<img width="467" alt="image" src="https://github.com/nextcloud/all-in-one/assets/106260515/411cd3a8-e3aa-4b1e-a05c-f67d5b73a4ff">

I'm unsure if the issue is with the High Performance Backend? 

Another thing I can add is that in Talk Settings in Administration, there is a red icon when clicking the test button. These are all at default settings and I haven't modified anything besides deleting the high performance backend setting:
(Yes, this was red **before** I deleted high performance backend as well)

<img width="1141" alt="image" src="https://github.com/nextcloud/all-in-one/assets/106260515/4a43f10a-970a-455e-98aa-78afb2edd4cb">

Also, I am more than happy to provide logs! I'd love to have the high performance backend running as that seems how it is meant to be.

I am a docker noob so please hold my hand if I need to run specific docker commands to grab them for you.

Thank you so much team for ALL of your hard work and dedication to Nextcloud!

**Comment by ardinusawan** [comment]
In my case, adding this in my nginx config solve the issue
```
   .... // others config

    # Add a location block to handle WebSocket connections
    location /push/ws {
        proxy_pass http://php-handler;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }

    location /cool {
        proxy_pass http://php-handler;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
```

**Comment by sherman-yang** [comment]
I have the same exact install process and issues. 
I'm sure everything is correct,  but install cannot validate domain. 
Collabora and Talk containers cannot work. makes the nextcloud webpage too slow. open a page takes 10 minutes.

**Reply by szaimen** [reply]
Hi, cna you follow https://github.com/nextcloud/all-in-one/discussions/1358?
